cg/services/file_delivery/fetch_file_service/fetch_delivery_files_service.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import Workflow
+from cg.constants.delivery import PIPELINE_ANALYSIS_TAG_MAP
+from cg.services.file_delivery.fetch_file_service.models import DeliveryFiles
+from cg.store.store import Store
+
+
+class FetchDeliveryFilesService(ABC):
+    """
+    Abstract class that encapsulates the logic required for managing the files to deliver.
+    1. Get a workflow and data delivery option
+    2. Fetch the files based on tags from Housekeeper
+    3. Return the files to deliver
+    """
+
+    @abstractmethod
+    def __init__(self, status_db: Store, hk_api: HousekeeperAPI):
+        self.status_db = status_db
+        self.hk_api = hk_api
+
+    @abstractmethod
+    def get_files_to_deliver(self, case_id: str) -> DeliveryFiles:
+        """Get the files to deliver."""
+        pass
+
+    @staticmethod
+    def get_analysis_sample_tags_for_workflow(workflow: Workflow) -> list[set[str]]:
+        return PIPELINE_ANALYSIS_TAG_MAP[workflow]["sample_tags"]
+
+    @staticmethod
+    def get_analysis_case_tags_for_workflow(workflow: Workflow) -> list[set[str]]:
+        return PIPELINE_ANALYSIS_TAG_MAP[workflow]["case_tags"]
```

Comment:
I really like this interface 🌞 🚀 

However, would it be possible to skip the `get_analysis_sample_tags_for_workflow` and `get_analysis_case_tags_for_workflow`? They seem like very specific methods that might not need to be part of the interface but rather hidden away somewhere internally.

It might make a lot of sense to have them here though, it just made me hesitant.

Reply:
Alternatively, it seems like those two methods might make more sense to put in a separate abstract class? A Python class can inherit from multiple abstract classes, so it might be better to split it into two separate ones since the functionality is quite different (get files to deliver vs get case/sample tags).

Reply:
Also, just for the record! This tag map is to be removed and deprecated: https://github.com/Clinical-Genomics/hermes/issues/102. In the future we'll only use one tag, `clinical-delivery` to extract the files we want to rsync

Reply:
Thanks for the info @ivadym. And agreed @seallard, I will split it off!